src/chain/network_event.rs

```diff
@@ -10,11 +10,37 @@ use super::{ProofSet, ProvingSection, SectionInfo};
 use crate::id::PublicId;
 use crate::parsec;
 use crate::sha3::Digest256;
+use crate::types::MessageId;
 use crate::{Authority, RoutingError, XorName};
 use hex_fmt::HexFmt;
 use maidsafe_utilities::serialisation::serialise;
 use std::fmt::{self, Debug, Formatter};
 
+#[derive(Debug, Ord, PartialOrd, Eq, PartialEq, Clone, Hash, Serialize, Deserialize)]
+pub struct ExpectCandidate {
+    /// The joining node's current public ID.
+    pub old_public_id: PublicId,
+    /// The joining node's current authority.
+    pub old_client_auth: Authority<XorName>,
+    /// The message's unique identifier.
+    pub message_id: MessageId,
+}
+
+#[derive(Debug, Serialize, Deserialize, PartialEq, Eq, PartialOrd, Ord, Hash, Clone)]
+pub struct ExpectCandidateVote {
+    // The routing_msg.content
+    pub rpc: ExpectCandidate,
+    // The routing_msg.dst
+    pub dst_name: XorName,
+}
+
+impl ExpectCandidateVote {
```

Comment:
It makes passing it around a lot easier though, and I do pass it around.

Reply:
yeh its fair play if we pass it around / you guys prefer it, but the name still is kind of weird considering its NetworkEvent we vote for in parsec from routing and this is a type within that enum and also called `..Vote`.

In this case the impl is also sorta weird to me cos we only use `into_network_event` I think once and if we constructed it directly, could have saved this entire impl definition. I'm prolly just biased to more primitive types that way instead of wrapper code.